README.md

```diff
@@ -1,50 +1,86 @@
+
+```sh
+kubectl create deploy/kubernetes/deployment.yaml
 ```
 
-# Test
+The external-provisioner may run in the same pod with other external CSI controllers such as the external-attacher, external-snapshotter and/or external-resizer.
 
-## Start Kubernetes
+Note that the external-provisioner does not scale with more replicas. Only one external-provisioner is elected as leader and running. The others are waiting for the leader to die. They re-elect a new active leader in ~15 seconds after death of the old leader.
 
-Run a local kubernetes cluster built from latest master code
+### Command line options
 
-## Run Storage Provider
+#### Important optional arguments that are highly recommended to be used
+* `--csi-address <path to CSI socket>`: This is the path to the CSI driver socket inside the pod that the external-provisioner container will use to issue CSI operations (`/run/csi/socket` is used by default).
 
-### Use HostPath drivers
+* `--enable-leader-election`: Enables leader election. This is useful when there are multiple replicas of the same external-provisioner running for one CSI driver. Only one of them may be active (=leader). A new leader will be re-elected when current leader dies or becomes unresponsive for ~15 seconds.
 
-Go to drivers and run:
+* `--timeout <duration>`: Timeout of all calls to CSI driver. It should be set to value that accommodates majority of `ControllerCreateVolume` and `ControlerDeleteVolume` calls. See [CSI error and timeout handling](#csi-error-and-timeout-handling) for details. 15 seconds is used by default.
 
-```bash
-bin/hostpathplugin --drivername mydriver  --endpoint unix://tmp/csi.sock --nodeid foobar -v=5
-```
+* `--retry-interval-start <duration>` - Initial retry interval of failed provisioning or deletion. It doubles with each failure, up to `--retry-interval-max` and then it stops increasing. Default value is 1 second. [CSI error and timeout handling](#csi-error-and-timeout-handling) for details.
 
-### Start external provisioner
+* `--retry-interval-max <duration>` - Maximum retry interval of failed provisioning or deletion. See [CSI error and timeout handling](#csi-error-and-timeout-handling) for details.
 
-```bash
-bin/csi-provisioner --kubeconfig /var/run/kubernetes/admin.kubeconfig --alsologtostderr
-```
+* `--worker-threads <num>`: Number of simultaneously running `ControllerCreateVolume` and `ControllerDeleteVolume` operations. Default value is `100`.
 
-### Create Storage class, PVC, and Pod
+#### Other recognized arguments
+* `--provisioning-retry-count <num>`: Sets number of retries of failed provisioning of a single volume. The external-provisioner will give up provisioning after given number of retries. It should be used only with special CSI drivers whose provisioning of volumes is extremely slow or expensive so the external-provisioner does not waste resources. Set to `0` to retry forever (which is also the default value).
```

Comment:
can you explain in more detail (at least to me) the use case of why you would give up rather than reconcile forever?

Reply:
I tried to track where the parameter comes from and found only this commit:

https://github.com/kubernetes-incubator/external-storage/commit/5277d96aa7d72bf157d329021e9b5fe7511e7d20

I can't find PR where it was discussed. Apparently some storage finds it better to stop provisioning instead of trying forever. @wongma7, do you remember?

Reply:
I am sure somebody requested/added it and I didn't object, I can't find the PR either. csi provisioner doesn't necessarily have to expose it

Reply:
OK, these parameters were not released, so I am removing them from the doc and I'll send a separate PR to remove them from the code.

Reply:
https://github.com/kubernetes-csi/external-provisioner/pull/254